app/models/concerns/episode_media.rb

```diff
@@ -111,39 +192,119 @@ def video_content_type?(*args)
-    media.inject(0) { |s, c| s + c.file_size.to_i }
+    if override?
+      external_media_resource&.file_size
+    else
+      media.inject(0) { |s, c| s + c.file_size.to_i }
+    end
   end
 
+  # 1) must_be_complete or not
+  # that is called with "true" in 2 places
+  # - in checking valid on 1st time publish
+  # - displaying "complete" for status
+  # So really this is about 1st time, must be fully ready,
+  # as there is no prior published media version to fall back on?
+  # And for status display or api responses, are media in the current version done?
+  #
+  # 2) override or not
+  # If there is an override, that is the only status to check
+  # We need a (different?) method to check contents ready for media versions
+  # but if an override is provided, we don't look at uploaded media status (uncut or contents)
+  #
+  # 3) enough segments
+  # Do we have enough media files for the segment count?
+  # That should get checked regardless of must_be_complete
+  #
+  # 4) uncut present
+  # If this is not must_be_complete, then
+  # If a single file upload, is there file uploaded, that isn't marked deleted
+  #
+  # 5) TODO: invalid or error status?
+  # should counts of files include invalid/error status files?
+  # should it be ready if any files is in error or invalid state?
   def media_ready?(must_be_complete = true)
-    if media.empty?
+    if !must_be_complete && medium_uncut?
+      uncut.present? && !uncut.marked_for_destruction?
+    elsif media.empty?
       false
-    elsif must_be_complete && !media.all?(&:status_complete?)
+    elsif segment_count.nil? && (media.size < media.map(&:position).max)
       false
-    elsif segment_count.nil?
-      media.size == media.map(&:position).max
+    elsif media.size < segment_count.to_i
+      false
+    elsif must_be_complete
+      media.all?(&:status_complete?)
     else
-      media.size >= segment_count
+      true
     end
   end
 
   def media_status
-    states = media.map(&:status).uniq
+    states = if override?
+      [external_media_resource&.status]
+    else
+      media.map(&:status).uniq
+    end
+
     if !(%w[started created processing retrying] & states).empty?
       "processing"
     elsif states.any? { |s| s == "error" }
       "error"
     elsif states.any? { |s| s == "invalid" }
       "invalid"
-    elsif media_ready?
+    elsif enclosure_ready?(true)
       "complete"
+    else
+      "incomplete"
     end
   end
 
   def media_url
     media.first.try(:href)
   end
+
+  def override_ready?(must_be_complete = true)
+    if override?
+      if must_be_complete
+        external_media_ready?(must_be_complete)
+      else
+        !enclosure_override_url.blank?
+      end
+    else
+      false
+    end
+  end
+
+  def override_processing?
+    !enclosure_override_url.blank? && !external_media_ready?
+  end
+
+  def override?
+    medium_override? || !enclosure_override_url.blank?
+  end
+
+  def external_media_ready?(must_be_complete = true)
```

Comment:
Is this still necessary, with `def override_ready?` being the main place we check?  I'm not sure I follow the distinction between the override being ready and the external media.

Reply:
a little duplicative, I'll combine it